bundles/org.openhab.binding.folderwatcher/src/main/java/org/openhab/binding/folderwatcher/internal/handler/FolderWatcherHandler.java

```diff
@@ -0,0 +1,257 @@
+
+                if (diff < config.diffHours) {
+                    file.setName(dirToList + "/" + currentFileName);
+                    dirList.add("ftp:/" + ftpClient.getRemoteAddress().toString() + file.getName());
+                }
+            }
+        }
+        return dirList;
+    }
+
+    private void connectionKeepAlive() {
+        if (ftp == null || ftp.isConnected() == false) {
+            if (config.secureMode.equals("NONE")) {
+                ftp = new FTPClient();
+            } else {
+                if (config.secureMode.equals("IMPLICIT")) {
+                    ftp = new FTPSClient(true);
+                } else if (config.secureMode.equals("EXPLICIT")) {
+                    ftp = new FTPSClient(false);
+                } else {
+                    ftp = null;
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Unsupported secure mode");
+                    return;
+                }
+            }
+
+            int reply = 0;
+            ftp.setListHiddenFiles(config.listHidden);
+            ftp.setConnectTimeout(config.connectionTimeout * 1000);
+
+            try {
+                if (config.ftpPort > 0) {
+                    ftp.connect(config.ftpAddress, config.ftpPort);
+                } else {
+                    ftp.connect(config.ftpAddress);
+                }
+                reply = ftp.getReplyCode();
+
+                if (!FTPReply.isPositiveCompletion(reply)) {
+                    ftp.disconnect();
+                    logger.debug("FTP server refused connection.");
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            "FTP server refused connection.");
+                    return;
+                }
+            } catch (IOException e) {
+                if (ftp.isConnected()) {
+                    try {
+                        ftp.disconnect();
+                    } catch (IOException e2) {
+                        logger.debug("Error disconneting, lost connection? : ", e2);
+                    }
+                }
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                return;
+            }
+            try {
+                if (!ftp.login(config.ftpUsername, config.ftpPassword)) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ftp.getReplyString());
+                    ftp.logout();
+                    return;
+                }
+                updateStatus(ThingStatus.ONLINE);
+                if (executionJob != null) {
+                    executionJob.cancel(true);
+                }
+                executionJob = scheduler.scheduleWithFixedDelay(this::refreshFTPFolderInformation,
+                        config.pollInterval + 5, config.pollInterval + 5, TimeUnit.SECONDS);
```

Comment:
why are you adding 5 to the configured poll interval? You shouldn't be adjusting configs like this since it goes against user expectation.